addon/components/base-layer.js

```diff
@@ -466,6 +486,27 @@ export default Ember.Component.extend(
     },
 
     /**
+      Returns leaflet layer's bounding box.
+
+      @method getBoundingBox
+      @return <a href="http://leafletjs.com/reference-1.1.0.html#latlngbounds">L.LatLngBounds</a>
+    */
+    getBoundingBox() {
+      let layer = this.get('_leafletObject');
+
+      if (Ember.isNone(layer)) {
+        return new Ember.RSVP.Promise((resolve, reject) => {
```

Comment:
for which pusposes here returns promise?

Reply:
Cuz private _getBoundingBox method returns a promise containing bbox, so the whole method should return a promise.

Reply:
ok